models/base/segment_web_page_views.sql

```diff
@@ -1,15 +1,50 @@
-with source as (
+with
 
-    select * from {{var('segment_page_views_table')}}
+{#
+The if statement below checks to see if segment_page_views_table is a string or a list, and then builds the model accordingly
+#}
+
+{% if var('segment_page_views_table') is string %}
+    
+    unioned_sources AS (
+        select 'segment_page_views_table' as source_name, * from {{var('segment_page_views_table')}}
+    ),
+
+
+{% elif var('segment_page_views_table') is iterable %}
+
+    {#
+    The section below takes each of the items listed for the segment_page_views_table variable, creates CTEs for them,
+    and then adds a field to note the name of the source table that the records are related to.
+    #}
+
+    {% for table_ref in var('segment_page_views_table', default=[]) %}
+        source_{{ loop.index }} as (
+            select '{{ table_ref }}' as source_name, *
+            from {{ ref(table_ref) }}
+        ),
+    {% endfor %}
+
+    {#
+    The section below takes all of the CTEs generated by the for loop above and then unions them together
+    #}
+
+        unioned_sources as (
+        {% for table_ref in var('segment_page_views_table', default=[]) %}
+            {% if loop.index > 1 %}union all{% endif %}
+            select * from source_{{ loop.index }}
+        {% endfor %}
```

Comment:
This union will require the same column set for all tables that are unioned, I think this is OK since users of the package can control what models they set here, but we may want to call that out somewhere.

Reply:
Good call, I updated the readme!